### ISSUE TITLE
Updates to a subscription in console are not persisted

### ISSUE THREAD

**Issue by maybeiambatman** [RetailMeNotSandbox/dart#7]
To reproduce:
1. Go to a subscription
2. Update the s3 path
3. Click save
4. Note success in upper right hand corner
5. View the subscription again (note the value is what you changed it to)
6. Refresh the page
7. View the subscription, one more time, and note that the value has reverted
Conclusion: changes to subscriptions are not persisted

Possible solutions:
* subscriptions are immutable or
* subscription changes are persisted and have the intended effect

**Comment by dannymcpherson** [comment]
If I recall correctly, this field was intended to be immutable.  If changes need to be made, the sanest way is to create a new subscription, and delete the old one.

**Comment by maybeiambatman** [comment]
I think there's a bigger issue here. Say we edit a subscription and change the s3 path. When we hit "Save" the api or the client won't complaint. When you refresh the page, you'll notice that the update hasn't gone through. Because of course, the field is immutable in the backend. However, the UI has deceived the user.

The solution here is having different schemas for "edit" and "create_new" modals. Seems easy enough. 

1. We probably have to change the generic definition of the ```getSchema()``` calls exposed by entities to accept a parameter e.g. ```getSchema(schema_type)``` where ```schema_type``` is "view", "create_new" etc. 

Or 

2. We can make specific functions for each use case e.g. ```getCreateNewSchema()```, ```getEditSchema()```

I personally like the first solution.

**Comment by maybeiambatman** [comment]
Creating a new issue (#12) for the additional schema work that needs to be done. Closing this ticket.

**Comment by maybeiambatman** [comment]
After reviewing the entities it seems like subscription is the only case affected by the issue. Instead of having to create entirely new separate schemas, I have opted for making a change on the client side (similar to how we set everything to read only when ```view```-ing an entity).

In case of subscription, when we see that the mode is ```edit``` we will set the following three fields to read only.

- ```s3_path_end_prefix_exclusive```
- ```s3_path_regex_filter```
- ```s3_path_start_prefix_inclusive```
